### PR TITLE
linter: add superlint for linting entire repo

### PR DESCRIPTION
**Pull request by gdams**
Adds https://github.com/marketplace/actions/super-linter to handle all the linting of relevant files in this repo.

A follow-up tasks would be to enable shellcheck and dockerfile linting (I've disabled them for now)

### CODE REVIEW

**Review by M-Davies — APPROVED** [review]
LGTM! Much nicer than Shellcheck. Hopefully this means https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/issues/14 can happen now 😉

**Comment by gdams** [comment]
Kind of concerning that our previous tests didn't catch https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/pull/37/files#diff-e13b21002fc6219ce1d9e3aff77c5c1cd1b2d4ea69b688af6c1f7d44b04c4fe9R87

**Comment by M-Davies** [comment]
> Kind of concerning that our previous tests didn't catch https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/pull/37/files#diff-e13b21002fc6219ce1d9e3aff77c5c1cd1b2d4ea69b688af6c1f7d44b04c4fe9R87

Our existing tests only lint [certain files](https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/blob/master/pipelines/src/test/groovy/TestCompilation.groovy#L67-L75) and the PR tester runs could have easily been missed amongst the large amount of subjobs you have to click through (plus we haven't actually ran tests for Power in a while so that map would usually never be read).

**Review by karianna — APPROVED** [review]
LGTM

File: pipelines/jobs/configurations/jdk13u_pipeline_config.groovy [hunk 1]
```diff
@@ -84,7 +84,7 @@ class Config13 {
                 arch                : 'ppc64',
                 additionalNodeLabels: 'xlc16',
                 test                : [
-                        nightly: []
+                        nightly: [],
```
Comment (karianna) [inline pipelines/jobs/configurations/jdk13u_pipeline_config.groovy:87]: @gdams you meant to put this back in?

**Pull request merged**